app/src/main/java/com/codesoom/assignment/controllers/AccountController.java

```diff
@@ -0,0 +1,100 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.application.AccountService;
+import com.codesoom.assignment.domain.Account;
+import com.codesoom.assignment.dto.AccountData;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+/**
+ *
+ * 웹에서 회원정보에 대한 요청을 받아 처리 및 출력을 담당합니다.
+ *
+ * @author flying_fisherman
+ */
+
+@RestController
+@CrossOrigin
+@RequestMapping("/products/accounts")
+public class AccountController {
+    private final AccountService accountService;
+
+    public AccountController(AccountService accountService) {
+        this.accountService = accountService;
+    }
+
+    /**
+     * 전체 회원정보를 List화 하여 리턴합니다.
+     *
+     * @return 전체 회원정보입니다.
+     */
+    @GetMapping
+    public List<Account> accountList() {
+        return accountService.getAccounts();
+    }
+
+    /**
+     * id를 받아 해당 id의 회원정보를 리턴합니다.
+     *
+     * @param id 조회하려는 회원정보 ID입니다.
+     * @return 해당 ID의 회원정보입니다.
+     */
```

Comment:
좋은 주석입니다. 그런데 `@param`에서 `id`에 대해 설명하고 있기 때문에 다음과 같이 축약할 수도 있어요.

```suggestion
    /**
     * 회원 정보를 리턴합니다.
     *
     * @param id 조회하려는 회원의 ID
     * @return 회원정보
     */
```

여기에서 생각해 볼 문제가 하나 더 있습니다. 만약에 존재하지 않는 id로 검색하려 한다면 이 메소드는 어떤 결과를 내놓아야 할까요?

Reply:
존재하지 않는 id로 검색하려 하는 경우, Service Layer의 `findAccount`를 통해 `AccountNotFoundException` 여부를 확인합니다.

JavaDoc 작성 시 거의 대부분의 메소드가 `findAccount`에 의존한다고 생각하여 `@throws` 태그는 해당 메소드에만 작성했습니다.